examples/fastify/src/index.ts

```diff
@@ -1,34 +1,57 @@
-import {
-  fastify,
-  FastifyReply,
-  FastifyRequest,
-  RouteShorthandOptions,
-} from "fastify";
+import { fastify, FastifyReply, FastifyRequest } from "fastify";
 import { autometrics } from "@autometrics/autometrics";
+import { PrismaClient } from "@prisma/client";
 
 const port = 7000;
 const server = fastify();
+const prisma = new PrismaClient();
 
-const opts: RouteShorthandOptions = {
-  schema: {
-    response: {
-      200: {
-        type: "object",
-        properties: {
-          body: {
-            type: "string",
-          },
-        },
-      },
-    },
-  },
-};
+interface Tulip {
+  id: number;
+  name: string;
+  price: number;
+}
 
-async function rootHandler(req: FastifyRequest, response: FastifyReply) {
-  return { body: "Hello world" };
+async function getAllTulips() {
+  return prisma.tulip.findMany();
 }
 
-server.get("/", opts, autometrics(rootHandler));
+async function handleGetAllTulips(req: FastifyRequest, res: FastifyReply) {
+  const tulips = await getAllTulips();
+  return {
+    data: tulips,
+  };
+}
+
+// Example internal function calling a database instrumented with autometrics
+const createTulip = autometrics(async function createTulip(
+  tulip: Tulip,
+) {
+  return prisma.tulip.create({
+    data: tulip,
+  });
+});
```

Comment:
I'd prefer the `createTulip` to be declared separately, which then can be wrapped. The intention is much clearer this way, and makes the comment redundant.

```ts
async function createTulip(tulip: Tulip) {
  return prisma.tulip.create({
    data: tulip,
  });
}

const createTulipAutometricsInstrumented = autometrics(createTulip);
```